crates/sui-core/src/execution_engine.rs

```diff
@@ -428,6 +428,17 @@ fn pay<S>(
                     Owner::AddressOwner(*recipient),
                     tx_ctx.digest(),
                 );
+                temporary_store.log_event(Event::TransferObject {
```

Comment:
don't we also need `NewObject` here?

Reply:
+1

Reply:
I will be replacing this with the new `CoinBalanceChange` event in next PR.